toolchain/check/decl_name_stack.h

```diff
@@ -237,10 +237,14 @@ class DeclNameStack {
   auto AddNameOrDiagnose(NameContext name_context, SemIR::InstId target_id,
                          SemIR::AccessKind access_kind) -> void;
 
-  // Adds a name to name lookup, or returns the existing instruction if this
+  // Adds a name to name lookup if not already declared or poisoned in this
```

Comment:
This phrasing makes it hard to tell whether "not" applies to "poisoned". I think this doesn't overwrite poisoned entries, so how about using "neither ... nor" to emphasize:

```suggestion
  // Adds a name to name lookup if neither already declared nor poisoned in this
```

If I'm wrong, swapping ordering could help:

```suggestion
  // Adds a name to name lookup if poisoned or not already declared in this
```

Or, "either" can also help make it clear "not" wouldn't apply to "poisoned":

```suggestion
  // Adds a name to name lookup if either not already declared or poisoned in this
```

Reply:
Done.